core/src/main/java/io/questdb/cutlass/line/CairoLineProtoParserSupport.java

```diff
@@ -57,136 +57,142 @@ public static void putValue(
-                    }
-                    row.putLong256(columnIndex, value, 2, limit);
-                    break;
-                case ColumnType.TIMESTAMP:
-                    row.putTimestamp(columnIndex, Numbers.parseLong(value, 0, value.length() - 1));
-                    break;
-                case ColumnType.CHAR:
-                    row.putChar(columnIndex, value.length() == 2 ? (char) 0 : value.charAt(1)); // skip quotes
-                    break;
-                case ColumnType.GEOBYTE:
-                    row.putByte(
-                            columnIndex,  // skip quotes
-                            (byte) GeoHashes.fromStringTruncatingNl(
-                                    value,
-                                    1,
-                                    value.length() - 1,
-                                    columnTypeMeta
-                            )
-                    );
-                    break;
-                case ColumnType.GEOSHORT:
-                    row.putShort(
-                            columnIndex,
-                            (short) GeoHashes.fromStringTruncatingNl(
-                                    value,
-                                    1,
-                                    value.length() - 1,
-                                    columnTypeMeta
-                            )
-                    );
-                    break;
-                case ColumnType.GEOINT:
-                    row.putInt(
-                            columnIndex,
-                            (int) GeoHashes.fromStringTruncatingNl(
-                                    value,
-                                    1,
-                                    value.length() - 1,
-                                    columnTypeMeta
-                            )
-                    );
-                    break;
-                case ColumnType.GEOLONG:
-                    row.putLong(
-                            columnIndex,
-                            GeoHashes.fromStringTruncatingNl(
-                                    value,
-                                    1,
-                                    value.length() - 1,
-                                    columnTypeMeta
-                            )
-                    );
-                    break;
-                default:
-                    // unsupported types are ignored
-                    break;
+        if (value.length() > 0) {
+            try {
+                switch (ColumnType.tagOf(columnType)) {
+                    case ColumnType.LONG:
+                        row.putLong(columnIndex, Numbers.parseLong(value, 0, value.length() - 1));
+                        break;
+                    case ColumnType.BOOLEAN:
+                        row.putBool(columnIndex, isTrue(value));
+                        break;
+                    case ColumnType.STRING:
+                        row.putStr(columnIndex, value, 1, value.length() - 2);
```

Comment:
what would happen here if length() == 1  ?

Reply:
this method is called as part of a two prong operation. First we call `getValueType`, which assigns the token a type, then we call `putValue` switching on the assigned type. For a value to be a string it has to be quoted, so a length of 2 is ensured:

```
case '"':
                    if (valueLen < 2 || value.charAt(0) != '\"') {
                        LOG.error().$("incorrectly quoted string: ").$(value).$();
                        return ColumnType.UNDEFINED;
                    }
                    return ColumnType.STRING;
```